### DISCUSSION TITLE
Run single test in vscode

### DISCUSSION THREAD

**Discussion by maciejbak85** [scalameta/metals#2972] [Q&A]
Hi,

is there a way to run single single test in vscode ?
Like in IntelliJ, basically to have an option from ide to run command like:

"sbt:projectname> it:testOnly *SomeTest -- -t "test description""

it is possible to configure a task of course, but is is possible to automate it ? Instead of configuring those tasks manually ?

**Comment by dos65** [comment]
Hi! Unfortunately, it isn't supported yet.
I will move it to feature-requests.

**Reply by maciejbak85** [reply]
would be great to see it, thanks

**Comment by ches** [comment]
VS Code has [recently been adding Test Explorer APIs](https://jpearson.blog/2021/09/01/test-explorer-in-visual-studio-code/), it'd be great to lay out what a framework would consist of for Metals to expose this with test case discovery defined for multiple test frameworks.

**Reply by tgodzik** [reply]
@kpodsiad is already taking a look at that actually in https://github.com/scalameta/metals/pull/3277 - that will not yet work for single tests, but for entire suites as a first step.
